client/graphviz.js

```diff
@@ -83,34 +89,41 @@
         } else if (ir.match(/^[A-Z]/)) {
           console.log('eval',ir)
 
-          if (ir.match(/^DOT (strict )?digraph/)) {
+          if (ir.match(/^DOT (strict )?(di)?graph/)) {
             context.graph = ir.replace(/DOT /,'')
-          }
+          } else
 
           if (ir.match(/^LINKS/)) {
             let text = context.want.map(p=>p.text).join("\n")
             let links = (text.match(/\[\[.*?\]\]/g)||[]).map(l => l.slice(2,-2))
             let tree = nest()
             links.map((link) => {
-              if (ir.match(/^LINKS HERE -> NODE/)) {
-                dot.push(`${quote(context.name)} -> ${quote(link)}`)
-              }
+              if (ir.match(/^LINKS HERE (->|--) NODE/)) {
+                let kind = ir.match(/->|--/)
+                dot.push(`${quote(context.name)} ${kind} ${quote(link)}`)
+              } else
               if (ir.match(/^LINKS NODE -> HERE/)) {
                 dot.push(`${quote(link)} -> ${quote(context.name)}`)
+              } else
+              if (!ir.match(/^LINKS$/)) {
+                trouble("can't do link", ir)
               }
               deeper.push({tree, context:Object.assign({},context,{name:link})})
             })
-          }
+          } else
 
           if (ir.match(/^HERE/)) {
             let tree = nest()
             let page = await get(context)
             if (page) {
               if (ir.match(/^HERE NODE/)) {
                 dot.push(quote(context.name))
-              }
+              } else
               if (ir.match(/^HERE NODE \w+/)) {
                 dot.push(`${quote(ir)} -> ${quote(context.name)} [style=dotted]`)
```

Comment:
digraph only?

Reply:
You are right. Here and everywhere else we emit dot format links we need to remember whether or not we were asked to generate a directed graph. I will make sure this information is recorded in the context and appropriately substituted in the places you have identified. Commit coming soon.

Reply:
Commit has arrived. See 4ac631e.